syntax = "proto3";

package ecommerce.ml;

option java_multiple_files = true;
option java_package = "com.intellicart.grpc.ml";

// The Recommendation service definition.
service RecommendationService {
  // Returns a list of personalized products based on user behavior
  rpc GetPersonalizedProducts (RecommendationRequest) returns (RecommendationResponse);
}

message RecommendationRequest {
  string user_id = 1;
  int32 max_results = 2;
  repeated string interested_categories = 3;
}

message ProductDetail {
  string product_id = 1;
  float score = 2;
  string reason = 3; // e.g., "Frequently bought together"
}

message RecommendationResponse {
  repeated ProductDetail products = 1;
  string model_version = 2;
}